assets/js/controllers/send.controller.js

```diff
@@ -391,6 +391,30 @@ function SendCtrl ($scope, $log, Wallet, Alerts, currency, $uibModal, $uibModalI
     return $q.resolve();
   };
 
+  $scope.handlePaste = (event) => {
+    let dest;
+
+    event.target.value
+      ? dest = event.target.value
+      : dest = $scope.transaction.destinations[0].address;
+
+    $timeout(() => {
+      event.target.value
+        ? dest = event.target.value
+        : dest = $scope.transaction.destinations[0].address;
+
+      function extractFromUri (URI, RegEx, sliceAmount) {
+        const data = URI.match(RegEx)[0];
+        return data.slice(sliceAmount, data.length);
+      }
+
+      $scope.transaction.destinations[0].address = extractFromUri(dest, /(?=\:)(.*)(?=\?)/, 1);
+      $scope.transaction.amounts[0] = extractFromUri(dest, /(?=amount=)(.*)(?=&message)/, 7) * 100000000;
+      $scope.transaction.note = decodeURI(extractFromUri(dest, /(?=&message=)(.*)/, 9));
```

Comment:
It might be better to process the URL in a helper method, which checks if it has the right format and extracts the address, amounts, and note. That helper method would return something like `{amount: 1, note: "Some note"}` to the controller. That keeps the controller simple.